sgt-back-end/index.js

```diff
@@ -0,0 +1,140 @@
+const express = require('express');
+const app = express();
+app.use(express.json());
+const pg = require('pg');
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+    select *
+      from "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      const grade = result.rows;
+      res.status(200).json(grade);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'Query has failed.'
+      });
+    });
+});
+
+app.post('/api/grades', (req, res) => {
+  const data = req.body;
+  const sql = {
+    text: 'INSERT INTO grades(name, course, score) VALUES($1, $2, $3) RETURNING *',
+    values: [data.name, data.course, data.score]
+  };
+  if (!data.name || !data.course || !data.course) {
+    res.status(400).json({
+      error: 'missing name, course, or score'
+    });
+    return;
+  }
+  if (data.score < 0) {
+    res.status(400).json({
+      error: 'score is not an integer from 0 to 100'
+    });
+    return;
+  }
+  db.query(sql)
+    .then(result => {
+      const grade = result.rows;
```

Comment:
Here `grade` will be an array with one grade object inside it. We only want to send the grade object back without the array. The goal of the server is to send back exactly what the client needs and can easily use so we don't want to send the single object inside an array in this case.

```JAVASCRIPT
const grade = result.rows[0];
```